### PR TITLE
Montgomery County, MD Local EITC 

### LINKED ISSUE #1010: Montgomery County, MD Local EITC
# Variable to add: local_eitc
Montgomery county, MD local EITC

It's a weird one, this is the only county.

### PR DESCRIPTION
**Pull request by YichunRen**
FIX #1010

### CODE REVIEW

File: policyengine_us/variables/gov/states/md/tax/income/credits/eitc/md_eitc.py [hunk 1]
```diff
@@ -11,4 +11,17 @@ class md_eitc(Variable):
...
+
+        # check montgomery county match qualification
+        county = tax_unit.household("county", period)
+        montgomery = county == "MONTGOMERY_COUNTY_MD"
+
+        # state EITC
+        refundable_eitc = tax_unit("md_refundable_eitc", period)
+        non_refundable_eitc = tax_unit("md_non_refundable_eitc", period)
+
+        return non_refundable_eitc + refundable_eitc(1 + p.match * montgomery)
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/md/tax/income/credits/eitc/md_eitc.py:27]: could you split the formula into multiple steps?

File: policyengine_us/variables/gov/states/md/tax/income/credits/eitc/md_eitc.py [hunk 2]
```diff
@@ -11,4 +11,17 @@ class md_eitc(Variable):
     reference = "https://casetext.com/statute/code-of-maryland/article-tax-general/title-10-income-tax/subtitle-7-income-tax-credits/section-10-704-effective-until-6302023-for-earned-income"
     defined_for = StateCode.MD
 
-    adds = ["md_non_refundable_eitc", "md_refundable_eitc"]
+    # adds = ["md_non_refundable_eitc", "md_refundable_eitc"]
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/md/tax/income/credits/eitc/md_eitc.py:14]:
```suggestion
```

File: policyengine_us/parameters/gov/states/md/tax/income/credits/eitc/montgomery/match.yaml [hunk 3]
```diff
@@ -0,0 +1,11 @@
+description: Montgomery County matches this percentage of the State Earned Income Tax Credit for taxpayers who qualify for the Refundable EITC from the State of Maryland, called the Working Families Income Supplement.
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/md/tax/income/credits/eitc/montgomery/match.yaml:1]:
```suggestion
description: Montgomery County matches this percentage of the refundable Maryland Earned Income Tax Credit.
```

File: policyengine_us/parameters/gov/states/md/tax/income/credits/eitc/montgomery/match.yaml [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+values:
+
+  2023-07-01: 0.56 # what date??
+metadata:
+  unit: /1
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/md/tax/income/credits/eitc/montgomery/match.yaml:6]:
```suggestion
  unit: /1
  period: year
```

File: policyengine_us/parameters/gov/states/md/tax/income/credits/eitc/montgomery/match.yaml [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+  label: Montgomery Refundable State EITC match
+  reference: # law reference??
+    - title: Montgomery County Earned Income Credit EIC Refund
+      href: https://www3.montgomerycountymd.gov/311/Solutions.aspx?SolutionId=1-4DAM0I
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/md/tax/income/credits/eitc/montgomery/match.yaml:10]: 2016 and prior law:
https://codelibrary.amlegal.com/codes/montgomerycounty/latest/montgomeryco_md/0-0-0-129735

It seems like it was not updated with the new bill 
I am reading this as a full match but 90% in 2025 and 95% in 2026 


But it definitely is 56% starting in 2023/07 - need to find more up to date bill

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: policyengine_us/variables/gov/states/md/tax/income/credits/eitc/refundable/montgomery_refundable_eitc.py [hunk 6]
```diff
@@ -0,0 +1,27 @@
+from policyengine_us.model_api import *
+
+
+class montgomery_refundable_eitc(Variable):
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/md/tax/income/credits/eitc/refundable/montgomery_refundable_eitc.py:4]:
```suggestion
class md_montgomery_eitc(Variable):
```

File: policyengine_us/variables/gov/states/md/tax/income/credits/eitc/refundable/montgomery_refundable_eitc.py [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+    value_type = float
+    entity = TaxUnit
+    label = "Montgomery EITC"
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/md/tax/income/credits/eitc/refundable/montgomery_refundable_eitc.py:7]:
```suggestion
    label = "Maryland montgomery county earned income tax credit"
```

File: policyengine_us/variables/gov/states/md/tax/income/credits/eitc/refundable/montgomery_refundable_eitc.py [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+    defined_for = StateCode.MD
+
+    def formula(tax_unit, period, parameters):
+        p = parameters(
+            period
+        ).gov.states.md.tax.income.credits.eitc.refundable.montgomery
+
+        # check montgomery county match qualification
+        county = tax_unit.household("county_str", period)
+        is_montgomery_resident = county == "MONTGOMERY_COUNTY_MD"
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/md/tax/income/credits/eitc/refundable/montgomery_refundable_eitc.py:21]:
```suggestion
        in_montgomery = county == "MONTGOMERY_COUNTY_MD"
```

File: policyengine_us/variables/gov/states/md/tax/income/credits/eitc/refundable/montgomery_refundable_eitc.py [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+
+        # state EITC
+        md_refundable_eitc = tax_unit("md_refundable_eitc", period)
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/md/tax/income/credits/eitc/refundable/montgomery_refundable_eitc.py:24]:
```suggestion
        state_refundable_eitc = tax_unit("md_refundable_eitc", period)
```

File: policyengine_us/parameters/gov/states/md/tax/income/credits/eitc/refundable/montgomery/match.yaml [hunk 10]
```diff
@@ -0,0 +1,18 @@
+description: Montgomery County matches this percentage of the refundable Maryland Earned Income Tax Credit. 
+values:
+  2015-01-01: 0.9
+  2016-01-01: 0.95
+  2020-01-01: 1
+  2023-07-01: 0.56 
+metadata:
+  unit: /1
+  period: year  
+  label: Montgomery Refundable State EITC match
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/md/tax/income/credits/eitc/refundable/montgomery/match.yaml:10]:
```suggestion
  label: Maryland montgomery refundable state earned income tax credit match
```

File: changelog_entry.yaml [hunk 11]
```diff
@@ -0,0 +1,5 @@
+- bump: minor
+  changes:
+    added:
+    - Montgomery County, MD Local EITC
```
Comment (PavelMakarchuk) [inline changelog_entry.yaml:4]:
```suggestion
    - Montgomery County Local EITC.
```

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: policyengine_us/parameters/gov/states/md/tax/income/credits/eitc/refundable/montgomery/match.yaml [hunk 12]
```diff
@@ -0,0 +1,18 @@
...
+  2015-01-01: 0.9
+  2016-01-01: 0.95
+  2020-01-01: 1
+  2023-07-01: 0.56 
+metadata:
+  unit: /1
+  period: year  
+  label: Maryland montgomery refundable state earned income tax credit match
+  reference: 
+    - title: 2023 Montgomery County Earned Income Credit EIC Refund
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/md/tax/income/credits/eitc/refundable/montgomery/match.yaml:12]: could add comments to the individual sources adding e.g. that the 2023 change is not yet reflected in the legal code

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: policyengine_us/parameters/gov/states/md/tax/income/credits/eitc/refundable/montgomery/match.yaml [hunk 13]
```diff
@@ -0,0 +1,26 @@
+description: Montgomery County matches this percentage of the refundable Maryland Earned Income Tax Credit.
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/md/tax/income/credits/eitc/refundable/montgomery/match.yaml:1]:
```suggestion
description: Montgomery County matches this percentage of the refundable Maryland earned income tax credit. 
```

File: policyengine_us/parameters/gov/states/md/tax/income/credits/eitc/refundable/montgomery/match.yaml [hunk 14]
```diff
@@ -0,0 +1,26 @@
+description: Montgomery County matches this percentage of the refundable Maryland Earned Income Tax Credit. 
+values:
+  2015-01-01: 0.9
+  2016-01-01: 0.95
+  2020-01-01: 1
+  2023-07-01: 0.56 
+metadata:
+  unit: /1
+  period: year  
+  label: Maryland montgomery refundable state earned income tax credit match
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/md/tax/income/credits/eitc/refundable/montgomery/match.yaml:10]:
```suggestion
  label: Maryland montgomery county refundable state earned income tax credit match
```

**Review by PavelMakarchuk — APPROVED** [review]

**Review by MaxGhenis — CHANGES_REQUESTED** [review]
please move these to the `local` folder(s) - see nyc as an example

File: policyengine_us/parameters/gov/states/md/tax/income/credits/eitc/refundable/montgomery/match.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/md/tax/income/credits/eitc/refundable/montgomery/match.yaml:1]: move to local

File: policyengine_us/variables/gov/states/md/tax/income/credits/eitc/refundable/montgomery_refundable_eitc.py
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/md/tax/income/credits/eitc/refundable/montgomery_refundable_eitc.py:1]: move to local

File: policyengine_us/tests/policy/baseline/gov/states/md/tax/income/credits/eitc/montgomery_refundable_eitc.yaml
Comment (PavelMakarchuk) [inline policyengine_us/tests/policy/baseline/gov/states/md/tax/income/credits/eitc/montgomery_refundable_eitc.yaml:1]: move to local

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: policyengine_us/parameters/gov/local/md/montgomery/tax/income/credits/refundable.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/local/md/montgomery/tax/income/credits/refundable.yaml:1]: delete

File: policyengine_us/variables/gov/local/md/montgomery/tax/income/credits/eitc/refundable/montgomery_refundable_eitc.py [hunk 15]
```diff
@@ -0,0 +1,27 @@
...
+    unit = USD
+    documentation = "Refundable Montgomery county EITC"
+    definition_period = YEAR
+    reference = "https://www3.montgomerycountymd.gov/311/Solutions.aspx?SolutionId=1-4DAM0I"
+    defined_for = StateCode.MD
+
+    def formula(tax_unit, period, parameters):
+        p = parameters(
+            period
+        ).gov.local.md.montgomery.tax.income.credits.eitc.refundable  # policyengine_us/parameters/gov/local/md/montgomery/tax/income/credits/eitc/refundable
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/local/md/montgomery/tax/income/credits/eitc/refundable/montgomery_refundable_eitc.py:17]:
```suggestion
        ).gov.local.md.montgomery.tax.income.credits.eitc.refundable  
```

File: policyengine_us/variables/gov/local/md/montgomery/tax/income/credits/eitc/refundable/montgomery_refundable_eitc.py [hunk 16]
```diff
@@ -0,0 +1,27 @@
...
+
+        # check montgomery county match qualification
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/local/md/montgomery/tax/income/credits/eitc/refundable/montgomery_refundable_eitc.py:19]:
```suggestion
        # Check whether unit is in montgomery county 
```

File: policyengine_us/parameters/gov/local/md/montgomery/tax/income/credits/eitc/refundable/match.yaml [hunk 17]
```diff
@@ -0,0 +1,26 @@
+description: Montgomery County matches this percentage of the refundable Maryland earned income tax credit. 
+values:
+  2015-01-01: 0.9
+  2016-01-01: 0.95
+  2020-01-01: 1
+  2023-07-01: 0.56 
+metadata:
+  unit: /1
+  period: year  
+  label: Montgomery County, Maryland EITC match
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/local/md/montgomery/tax/income/credits/eitc/refundable/match.yaml:10]:
```suggestion
  label: Montgomery County EITC match
```

File: policyengine_us/parameters/gov/local/md/montgomery/tax/income/credits/eitc/refundable/match.yaml [hunk 18]
```diff
@@ -0,0 +1,26 @@
+description: Montgomery County matches this percentage of the refundable Maryland earned income tax credit.
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/local/md/montgomery/tax/income/credits/eitc/refundable/match.yaml:1]:
```suggestion
description: Montgomery County matches this percentage of the refundable Maryland Earned Income Tax Credit. 
```

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

**Review by PavelMakarchuk — APPROVED** [review]

File: policyengine_us/variables/gov/local/md/montgomery/tax/income/credits/eitc/refundable/montgomery_refundable_eitc.py [hunk 19]
```diff
@@ -0,0 +1,27 @@
+from policyengine_us.model_api import *
+
+
+class md_montgomery_eitc(Variable):
+    value_type = float
+    entity = TaxUnit
+    label = "Maryland montgomery county earned income tax credit"
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/local/md/montgomery/tax/income/credits/eitc/refundable/montgomery_refundable_eitc.py:7]:
```suggestion
    label = "Montgomery County, Maryland EITC"
```

File: policyengine_us/variables/gov/local/md/montgomery/tax/income/credits/eitc/refundable/montgomery_refundable_eitc.py [hunk 20]
```diff
@@ -0,0 +1,27 @@
...
+    unit = USD
+    documentation = "Refundable Montgomery county EITC"
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/local/md/montgomery/tax/income/credits/eitc/refundable/montgomery_refundable_eitc.py:9]:
```suggestion
```

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**